sdk/servicebus/service-bus/test/atomManagement.spec.ts

```diff
@@ -22,12 +22,15 @@ import * as dotenv from "dotenv";
 dotenv.config();
 
 import { EnvVarKeys, getEnvVars } from "./utils/envVarUtils";
+import { getEndpointWithProtocolFromConnectionString } from "./utils/testUtils";
 const env = getEnvVars();
 
 const serviceBusAtomManagementClient: ServiceBusAtomManagementClient = new ServiceBusAtomManagementClient(
   env[EnvVarKeys.SERVICEBUS_CONNECTION_STRING]
 );
 
+const endpointWithProtocol = getEndpointWithProtocolFromConnectionString();
```

Comment:
We have access to the service bus connection string in the previous line, and here we are calling a utility function that will fetch the connection string by the means of `getEnvVars()` one more time.

Since this utility function `getEndpointWithProtocolFromConnectionString` is used only in a single place and since we already have access to the connection string, I would suggest to inline the code to fetch endpoint from the connection string.

Reply:
Idea was that since `endpointWithProtocol` is something that's being used to construct input values to testcases as well.
Was thinking there might be a case in future where we might want to split the tests into many files. In such cases, breaking down utilities to smaller consumable functions would help with modularity and maintenance.

Because parseConnectionString is a utility we define in amqp-common and it's signature for instance needs to be updated from `string` to something more appropriate. For now it's fine, but improvement to it should hopefully allow us to respond with updates to fewer places.

But given current state, sure we can optimize the test runs

Reply:
If you want to keep the utility function then you can change it such that it takes an optional string parameter (i.e. `connectionString`). If the value of `connectionString` is not specified, then retrieve it from the environment. Otherwise use the one passed in.

That way, in this code you will pass to the method the value of `env[EnvVarKeys.SERVICEBUS_CONNECTION_STRING]` without having to hit the environment again. 

In the `getEndpointWithProtocolFromConnectionString` method you can do something like:

```ts
export function getEndpointWithProtocolFromConnectionString(string connectionString): string {
   // if no connection string has been passed in, retrieve it from the environment
   if (connectionString === ""){
     const env = getEnvVars(); 
     connectionString = env[EnvVarKeys.SERVICEBUS_CONNECTION_STRING];
   }

   const connectionStringObj: any = parseConnectionString(
     connectionString
   );
   return connectionStringObj.Endpoint;
 }
```